### PR TITLE
feat(Currys): add presence

### PR DESCRIPTION
**Pull request by xFutuxe**
## Description 

Presence added for the website/retailer [Currys](https://www.currys.co.uk)
(closed old PR and made a new one. Needed to rework things, any old changed are shown on the closed pr)
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

proof1
proof2

</details>

### CODE REVIEW

File: websites/C/Currys/presence.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	if (location.href.includes("search")) {
+		// Searching product
+		const replaced = document
+			.querySelector(".plp-heading")
+			.textContent.replace(/["]/g, " ")
+			.replace("Showing results for", "");
+
+		presenceData.details = "Searching for:";
+		presenceData.state = replaced;
+	}
```
Comment (DarkVillager) [inline websites/C/Currys/presence.ts:67]: I'd change this.

First let's detect if there's anything on the search bar:
```ts
const search = document.querySelector<HTMLInputElement>('[id="Search"]')
If (search?.value) {
		presenceData.details = "Searching for:";
		presenceData.state= search.value
                presenceData.smallImageKey = "https://i.imgur.com/B5iUj20.png"
                return presence.setActivity(presenceData);

}
```

And to replace the current check here:
```ts
If (pathname === "search") {
		presenceData.details = "Viewing search results for:";
		presenceData.state= document.querySelector('[class="breadcrumb"]').lastElementChild?.textContent?.trim()
                presenceData.smallImageKey = "https://i.imgur.com/B5iUj20.png"
}
```

**Comment by DarkVillager** [comment]
> (closed old PR and made a new one. Needed to rework things, any old changed are shown on the closed pr)
Please don't close PRs unless you're abandoning the presence/rework or its corrupted.

**Comment by xFutuxe** [comment]
> Please don't close PRs unless you're abandoning the presence/rework or its corrupted.

mb.

**Review by Bas950 — APPROVED** [review]

File: websites/C/Currys/presence.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	// When searching for product
+	const search = document.querySelector<HTMLInputElement>('[id="Search"]');
+	if (search?.value) {
+		presenceData.details = "Searching for:";
+		presenceData.state = search.value;
+		presenceData.smallImageKey = "https://i.imgur.com/B5iUj20.png";
+		return presence.setActivity(presenceData);
+	}
+
+	if (pathname === "search") {
```
Comment (DarkVillager) [inline websites/C/Currys/presence.ts:66]: otherwise it won't work
```suggestion
	if (pathname === "/search") {
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/C/Currys/presence.ts (hunk 2 above)
Comment (xFutuxe) [reply websites/C/Currys/presence.ts:66]: I didn't spot that when checking over it. thanks

**Review by Timeraa — APPROVED** [review]

**Pull request merged**